docs/flow/config-language/syntax.md

```diff
@@ -6,3 +6,88 @@ weight: 200
 ---
 
 # Syntax
+The River syntax is designed to be easy to read and write. Essentially, there
+are just two high-level elements to it: _Attributes_ and  _Blocks_. 
+
+River is a _declarative_ language used to build programmable pipelines.
+As such, the ordering of blocks and attributes within the River configuration
+file is not important; the language will consider all direct and indirect
+dependencies between elements to determine their relationships.
+
+## Attributes and Blocks
+
+### Attributes
+_Attributes_ are used to configure individual settings. They always take the
+form of `<ATTRIBUTE_NAME> = <ATTRIBUTE_VALUE>`. They can appear either as
+top-level elements or nested within blocks.
+
+The following example  sets the `log_level` attribute to `"debug"`.
+
+```river
+log_level = "debug"
+```
+
+All `<ATTRIBUTE_NAME>`s must be valid River identifiers.
+
+The `<ATTRIBUTE_VALUE>` can be either a constant value of a valid River
+[type]({{< relref "./expressions/types_and_values.md" >}}) (eg. string,
+boolean, number) or an [_expression_]({{< relref "./expressions/_index.md" >}})
+to represent or compute more complex attribute values.
+
+### Blocks
+_Blocks_ are used to configure components by grouping any number of attributes
+or nested blocks using curly braces.
+
+Blocks have a _name_ (the type of the component it defines), an optional
+_label_ (depending on the component type) and a body that contains any number
+of arguments or nested unlabeled blocks.
+
+```
+// Pattern for creating an unlabeled block:
+<BLOCK NAME> {
+	// Block body can contain attributes and nested unlabeled blocks
+	<IDENTIFIER> = <EXPRESSION> // Attribute
+
+	<NESTED_BLOCK_NAME> {
+		// Nested block body
+	}
+}
+
+// Pattern for creating a labeled block:
+<BLOCK NAME> "<BLOCK LABEL>" {
+	// Block body can contain attributes and nested unlabeled blocks
+	<IDENTIFIER> = <EXPRESSION> // Attribute
+
+	<NESTED_BLOCK_NAME> {
+		// Nested block body
+	}
+}
+```
+
+The `<BLOCK_NAME>` must correspond to a registered Flow [component]({{< relref "./components.md" >}}).
```

Comment:
like above, this is not necessarily true: there are permissible top-level blocks like logging which are not components. 

This can probably say something like "the block names have to be recognized by Flow."